grapes.js

```diff
@@ -71,17 +73,25 @@ class Grapes extends EventEmitter {
   }
 
   addDefaultGrapes () {
-    const g1 = new Grape({
+    const { ports } = this.opts
+    const [inG1Ports, inG2Ports] = Array.isArray(ports)
+      ? ports
+      : []
+    const g1Ports = {
       dht_port: 20002,
       dht_bootstrap: ['127.0.0.1:20001'],
-      api_port: 40001
-    })
-    const g2 = new Grape({
+      api_port: 40001,
+      ...inG1Ports
```

Comment:
Not sure to follow the logic here, why not sending in opts ports the whole g1Ports or g2Ports and use the default values if nothing is send?